rest/client.go

```diff
@@ -156,6 +167,26 @@ func (c Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 	return resp, err
 }
 
+// nextFunc knows how to get and parse additional info from uri into v.
+type nextFunc func(v *interface{}, uri string) (*http.Response, error)
+
+// DoWithPagination Does, and follows Link headers for pagination.
+func (c Client) DoWithPagination(req *http.Request, v interface{}, f nextFunc) (*http.Response, error) {
```

Comment:
I recommend you use a pointer receiver here `func (c *Client) ...`

As it's currently set up, I don't think it would cause any problems, but you may find it helps avoid problems down the road if any of your fields become non-pointer structs.

Reply:
done